sdk/src/main/java/com/bugsnag/android/AppData.java

```diff
@@ -34,31 +41,36 @@
         this.appContext = appContext;
         this.sessionTracker = sessionTracker;
         appName = getAppName(appContext);
-        packageName = getPackageName(appContext);
+
+        packageName = calculatePackageName(appContext);
+        buildUuid = config.getBuildUUID();
+        durationMs = getDurationMs();
+        foregroundMs = sessionTracker.getDurationInForegroundMs(System.currentTimeMillis());
+        inForeground = sessionTracker.isInForeground();
     }
 
     @Override
     public void toStream(@NonNull JsonStream writer) throws IOException {
         writer.beginObject();
         serialiseMinimalAppData(writer);
-
         writer.name("id").value(packageName);
-        writer.name("buildUUID").value(config.getBuildUUID());
-        writer.name("duration").value(getDurationMs());
-        long foregroundMs = sessionTracker.getDurationInForegroundMs(System.currentTimeMillis());
+        writer.name("buildUUID").value(buildUuid);
+        writer.name("duration").value(durationMs);
         writer.name("durationInForeground").value(foregroundMs);
-        writer.name("inForeground").value(sessionTracker.isInForeground());
-
-        // TODO migrate legacy fields
-        writer.name("name").value(appName);
-        writer.name("packageName").value(packageName);
-        writer.name("versionName").value(versionName);
-        writer.name("activeScreen").value(getActiveScreenClass());
-        writer.name("memoryUsage").value(getMemoryUsage());
-        writer.name("lowMemory").value(isLowMemory(appContext));
+        writer.name("inForeground").value(inForeground);
         writer.endObject();
     }
 
+    // TODO migrate metadata fields to separate class
+    void addAppMetaData(MetaData metaData) {
+        metaData.addToTab("app", "name", appName);
+        metaData.addToTab("app", "packageName", packageName);
+        metaData.addToTab("app", "versionName", getVersionName());
+        metaData.addToTab("app", "activeScreen", getActiveScreenClass());
+        metaData.addToTab("app", "memoryUsage", getMemoryUsage());
+        metaData.addToTab("app", "lowMemory", isLowMemory(appContext));
```

Comment:
How would the unity notifier access these values?

Reply:
Comment addressed in #333